doc/adr004-effect-system.md

```diff
@@ -38,150 +36,292 @@ simplicity.
 
 ## Solution
 
-Quint expressions and definitions can be assigned both a type and an effect, which
-are independent of each other. Here are some examples of inferred effects for
-some definitions:
+Quint expressions and definitions can be assigned both a type and an effect,
+which are independent of each other. Here are some examples of inferred effects
+for some definitions:
 
 ```
 var x: int
 var y: int
 
-pure P(a) = a + 1 // (Pure) => Pure
-state S(a) = x + a // (Pure) => Read['x']
+pure def P(a) = a + 1 // (Read[r1] & Temporal[t1]) => Read[r1] & Temporal[t1]
+def S(a) = y + a // (Read[r1] & Temporal[t1]) => Read['y', r1] & Temporal[t1]
 
-action<x> A1(a) = x <- x + a // (Pure) => Read['x'] & Update['x']
-action<x> A2(a) = x <- S(a) // (Pure) => Read['x'] & Update['x']
-
-action<x, y> A3(a) = x <- x + a // (Pure) => Read['x', 'y'] & Update['x', 'y']
+action A1 = x' = x + 1 // Read['x'] & Update['x']
+action A2 = x' = S(1) // Read['y'] & Update['x']
 ```
 
-
-In these examples, keywords `pure`, `state` and `action` are providing effect
+In these examples, keywords `pure def`, `def` and `action` are providing effect
 information to be checked against the inferred effect. This information is not
-used for inference. Note that information between `<...>` is not part of the
-effect system, but actually syntax sugar for unchanged assignments (and
-assignments do have influence on effects).
+used for inference.
 
 ### How restrictive should `pure` definitions be?
 
-This is an open question. We have two alternatives for the `pure` keyword:
+We have two alternatives for the `pure` keyword:
+
 1. Only allow Pure parameters and Pure result
-2. Allow parameters with Read effects as long as the resulting effect doesn't Read
-   any other variables
+2. Allow parameters with Read effects as long as the resulting effect doesn't
+   read any other variables
 
 Option 1 may be too restrictive and barely used, since, for most operators, it's
 desirable that they can be applied to state variables (i.e. `+` in `x + 1` if
 `x` is a state variable).
 
-PS: alternative names for `pure` are `static` and `stateless`.
+### Decision
+
+We have decided to implement option 2 for all modes. According to this option, a
+definition is considered pure if it does not add any read effect on top of the
+effects that the parameters may carry. A definition is considered `def` if it
```

Comment:
```suggestion
effects that the parameters may carry. A definition is considered `pure` if it
```

Reply:
This is actually meant to be `def`. Since the option 2 was implemented for all modes, not only pure, I gave a description of the `def` mode as well. I'll update this to clarify that this is another example, as I agree it is a bit confusing in the current structure.